src/CMakeLists.txt

```
@@ -43,6 +43,22 @@ if(NOT (BUILD_CLIENT OR BUILD_SERVER))
 endif()
 
 
+option(PRECOMPILE_HEADERS "Precompile some headers (experimental)" FALSE)
```

The option should be limited to CMake 3.16+

---

I've added an error for 3.16-.

---

I meant like this:
```suggestion
if(NOT ${CMAKE_VERSION} VERSION_LESS 3.16) # VERSION_GREATER_EQUAL is CMake 3.7 while Minetest is at 3.5
	option(PRECOMPILE_HEADERS "Precompile some headers (experimental)" FALSE)
```
or maybe
```suggestion
option(PRECOMPILE_HEADERS "Precompile some headers (experimental; requires CMake 3.16 or later)" FALSE)
```
Simply adding an error is not very useful, it would error at `target_precompile_headers` anyway.

---

>Simply adding an error is not very useful

It's an error with a meaningful message though. So imo it is useful.

Added to doc that 3.16 is required